caravel/models.py

```diff
@@ -466,11 +494,25 @@ def import_obj(cls, dashboard_to_import, import_time=None):
         # copy slices object as Slice.import_slice will mutate the slice
         # and will remove the existing dashboard - slice association
         slices = copy(dashboard_to_import.slices)
-        slice_ids = set()
+        old_to_new_slc_id_dict = {}
+        new_filter_immune_slices = []
+        new_expanded_slices = {}
+        i_params_dict = dashboard_to_import.params_dict
         for slc in slices:
             logging.info('Importing slice {} from the dashboard: {}'.format(
                 slc.to_json(), dashboard_to_import.dashboard_title))
-            slice_ids.add(Slice.import_obj(slc, import_time=import_time))
+            new_slc_id = Slice.import_obj(slc, import_time=import_time)
+            old_to_new_slc_id_dict[slc.id] = new_slc_id
+            # update json metadata that deals with slice ids
+            if ('filter_immune_slices' in i_params_dict and
+                    slc.id in i_params_dict['filter_immune_slices']):
+                new_filter_immune_slices.append(new_slc_id)
+            new_slc_id_str = '{}'.format(new_slc_id)
+            old_slc_id_str = '{}'.format(slc.id)
+            if ('expanded_slices' in i_params_dict and
+                    old_slc_id_str in i_params_dict['expanded_slices']):
```

Comment:
http://stackoverflow.com/questions/1450957/pythons-json-module-converts-int-dictionary-keys-to-strings
Added workaround here.
@mistercrunch  ^^